planner/core/logical_plans.go

```diff
@@ -531,6 +546,40 @@ func (ds *DataSource) buildIndexGather(path *util.AccessPath) LogicalPlan {
 	return sg
 }
 
+func (ds *DataSource) buildIndexLookupGather(path *util.AccessPath, idxCols []*expression.Column, idxColLens []int) LogicalPlan {
+	is := LogicalIndexScan{
+		Source:       ds,
+		IsDoubleRead: true,
+		Index:        path.Index,
+	}.Init(ds.ctx, ds.blockOffset)
+	is.Columns = make([]*model.ColumnInfo, len(ds.Columns))
+	copy(is.Columns, ds.Columns)
+	is.SetSchema(ds.schema)
```

Comment:
IndexScan's schema should be the index columns.

Reply:
Done.